dwave/optimization/states.pyx

```diff
@@ -0,0 +1,331 @@
+# Copyright 2024 D-Wave
```

Comment:
Is `# distutils: language = c++` still recommended for cython files?

Reply:
`distutils` itself is long since deprecated. And ever since we switched to `meson` build system these headers are ignored. I intent to clear them out of the other files as I go.